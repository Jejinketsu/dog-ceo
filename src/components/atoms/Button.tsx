import { Text, Pressable, StyleProp, ViewStyle } from "react-native";
import React, { ComponentProps } from "react";
import Icons from "./icons";

type ButtonProps = {
  style?: StyleProp<ViewStyle>;
  children?: React.ReactNode;
  PressableProps?: ComponentProps<typeof Pressable>;
  IconProps?: ComponentProps<typeof Icons>;
};

const Button = (props: ButtonProps) => {
  const { children, style, PressableProps, IconProps } = props;
  return (
    <Pressable
      style={style}
      className="active:opacity-50 rounded-lg border border-gray-200 bg-white pt-2 pb-2 pl-4 pr-4"
      {...PressableProps}
    >
      {IconProps && <Icons {...IconProps} />}
      {children && <Text>{children}</Text>}
    </Pressable>
  );
};

export default Button;
