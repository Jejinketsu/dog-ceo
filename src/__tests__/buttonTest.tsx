import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/atoms/Button";

test("renders correctly", () => {
  const tree = renderer
    .create(<Button IconProps={{ name: "hearth", size: 24, color: "black" }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
