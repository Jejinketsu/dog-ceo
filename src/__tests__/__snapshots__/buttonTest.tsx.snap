// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "borderBottomColor": "#e5e7eb",
        "borderLeftColor": "#e5e7eb",
        "borderRightColor": "#e5e7eb",
        "borderTopColor": "#e5e7eb",
      },
      {
        "backgroundColor": "#fff",
      },
      {
        "paddingTop": 8,
      },
      {
        "paddingBottom": 8,
      },
      {
        "paddingLeft": 16,
      },
      {
        "paddingRight": 16,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24px"
    bbWidth="24px"
    fill="black"
    focusable={false}
    height="24px"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="24px"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
